components/tor/onion_location_navigation_throttle.cc

```diff
@@ -70,13 +70,17 @@ OnionLocationNavigationThrottle::WillProcessResponse() {
   // https://gitweb.torproject.org/tor-browser-spec.git/plain/proposals/100-onion-location-header.txt
   if (headers && GetOnionLocation(headers, &onion_location) &&
       !navigation_handle()->GetURL().DomainIs("onion")) {
+    GURL url(onion_location);
+    // The Onion-Location value must be a valid URL with http: or https:
+    // protocol and a .onion hostname.
+    if (!url.SchemeIsHTTPOrHTTPS() || !url.DomainIs("onion"))
+      return content::NavigationThrottle::PROCEED;
```

Comment:
++. we also need to check that the opener page (`navigation_handle()->GetURL()`?) protocol is `https://` only, unless i'm missing that somewhere

Reply:
fixed in 68c19e3f79ded962cbdd84d9835e44f104f854bf